utility/src/test/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecoratorTest.kt

```diff
@@ -0,0 +1,76 @@
+package org.oppia.android.util.profile
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.extensions.proto.LiteProtoTruth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.util.profile.CurrentUserProfileIdIntentDecorator.decorateWithUserProfileId
+import org.oppia.android.util.profile.CurrentUserProfileIdIntentDecorator.extractCurrentUserProfileId
+import org.robolectric.annotation.LooperMode
+
+/** Tests for [CurrentUserProfileIdIntentDecorator]. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class CurrentUserProfileIdIntentDecoratorTest {
+
+  @Test
+  fun testDecorator_decorateWithProfileId_returnsIntentWithCorrectProfileId() {
+    val intent = Intent().apply {
+      decorateWithUserProfileId(
+        ProfileId.newBuilder().apply {
+          internalId = 0
```

Comment:
I suggest using a non-zero ID (as you did below) since the default proto value is 0, so a non-zero value helps improve the test robustness and resilience against unintentional side effects of production or test arrangement changes.

Reply:
Done.